### PR TITLE
ios-binding and ios-demo

### PR DESCRIPTION
**Pull request by ksyeo1010**
- add leopard ios binding
- add ios demo
- add tests

### CODE REVIEW

**Comment by ErisMik** [comment]
Are the iOS libs in Release not Debug ;)

**Review by ErisMik — APPROVED** [review]

**Comment by ksyeo1010** [comment]
> Are the iOS libs in Release not Debug ;)

Release :)

File: demo/ios/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+To obtain your `AccessKey`:
+1. Login or Signup for a free account on the [Picovoice Console](https://picovoice.ai/console/).
+2. Once logged in, go to the [`AccessKey` tab](https://console.picovoice.ai/access_key) to create one or use an existing `AccessKey`.
+
+Copy your `AccessKey` into the `ACCESS_KEY` variable inside [`ViewModel.swift`](/demo/ios/LeopardDemo/LeopardDemo/ViewModel.swift) before building the demo.
+
+## Running the Demo
+
+Before building the demo app, run the following from `LeopardDemo` directory to install the Leopard Cocoapod:
```
Comment (kenarsa) [inline demo/ios/README.md:16]: `LeopardDemo` >>> [LeopardDemo](link ...)

File: demo/ios/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+```ruby
+pod install
+```
+Open `LeopardDemo/LeopardDemo.xcworkspace` and run the demo.
+
+## Running the On-Device Unit Tests
+
+Copy your `AccessKey` into the `accessKey` variable in `LeopardDemo/LeopardDemoUITests/LeopardDemoUITests.swift`. Open `LeopardDemo/LeopardDemo.xcworkspace` with XCode and run the tests with `Product > Test`.
```
Comment (kenarsa) [inline demo/ios/README.md:25]: [LeopardDemoUITests.swift]()

make things that can be linked to a link please

File: binding/ios/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    - Windows (x86_64)
+    - Raspberry Pi (4, 3)
+    - NVIDIA Jetson Nano
+    - Android and iOS
+
+## Installation
+
+The Leopard iOS binding is available via [Cocoapods](https://cocoapods.org/pods/Leopard-iOS). To import it into your iOS project, add the following line to your Podfile: 
+
+```ruby
```
Comment (kenarsa) [inline binding/ios/README.md:24]: console?

**Review by kenarsa — CHANGES_REQUESTED** [review]

File: binding/ios/README.md (hunk 3 above)
Comment (ksyeo1010) [reply binding/ios/README.md:24]: pod file is actually ruby

File: binding/ios/Leopard.swift [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+//  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+//  file accompanying this source.
+//  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+//  specific language governing permissions and limitations under the License.
+//
+
+import PvLeopard
+
+/// iOS binding for Leopard wake word engine. Provides a Swift interface to the Leopard library.
```
Comment (kenarsa) [inline binding/ios/Leopard.swift:12]: `wake word engine` 🧐

File: binding/ios/Leopard.swift [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    ///   - modelPath: Absolute path to file containing model parameters.
+    /// - Throws: LeopardError
+    public init(accessKey: String, modelPath: String) throws {
+
+        if accessKey.count == 0 {
+            throw LeopardInvalidArgumentError("AccessKey is required for Leopard initialization")
+        }
+
+        if !FileManager().fileExists(atPath: modelPath) {
+            throw LeopardInvalidArgumentError("Model file at does not exist at '\(modelPath)'")
```
Comment (kenarsa) [inline binding/ios/Leopard.swift:33]: remove extra `at`

File: binding/ios/Leopard.swift [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    /// Processes a given audio data and returns its transcription.
+    ///
+    /// - Parameters:
+    ///   - pcm: An array of 16-bit pcm samples. The audio needs to have a sample rate equal to `.sample_rate` and be 16-bit
+    ///          linearly-encoded. This function operates on single-channel audio. If you wish to process data in a different
+    ///          sample rate or format consider using `.process_file`.
+    /// - Throws: LeopardError
+    /// - Returns: Inferred transcription.
+    public func process(_ pcm:[Int16]) throws -> String {
+        if handle == nil {
```
Comment (kenarsa) [inline binding/ios/Leopard.swift:74]: is this even possible? just figuring if we need this check

File: binding/ios/README.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+- Private; All voice processing runs locally.
+- Accurate [[1]](https://github.com/Picovoice/speech-to-text-benchmark#results)
+- Compact and Computationally-Efficient [[1]](https://github.com/Picovoice/speech-to-text-benchmark#results)
+- Cross-Platform:
+    - Linux (x86_64)
+    - macOS (x86_64, arm64)
+    - Windows (x86_64)
+    - Raspberry Pi (4, 3)
+    - NVIDIA Jetson Nano
+    - Android and iOS
```
Comment (kenarsa) [inline binding/ios/README.md:18]: let's make the order the same as the main README.md

File: binding/ios/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+The Leopard SDK requires a valid `AccessKey` at initialization. `AccessKey`s act as your credentials when using Leopard SDKs.
+You can create your `AccessKey` for free. Make sure to keep your `AccessKey` secret.
+
+To obtain your `AccessKey`:
+1. Login or Signup for a free account on the [Picovoice Console](https://picovoice.ai/console/).
+2. Once logged in, go to the [`AccessKey` tab](https://console.picovoice.ai/access_key) to create one or use an existing `AccessKey`.
+
+## Usage
+
+Add the Leopard model file to `Xcode` and create an instance of the engine:
```
Comment (kenarsa) [inline binding/ios/README.md:39]: can we write a couple of sentences here about:

1- default is under lib/common
2- how to put it in the bundle (where? ) if it makes sense?
3- link to console to create their own ... obviously keep this vague for now.

File: demo/ios/README.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+1. Login or Signup for a free account on the [Picovoice Console](https://picovoice.ai/console/).
+2. Once logged in, go to the [`AccessKey` tab](https://console.picovoice.ai/access_key) to create one or use an existing `AccessKey`.
+
+Copy your `AccessKey` into the `ACCESS_KEY` variable inside [`ViewModel.swift`](/demo/ios/LeopardDemo/LeopardDemo/ViewModel.swift#L24) before building the demo.
+
+## Running the Demo
+
+Before building the demo app, run the following from [LeopardDemo](/demo/ios/LeopardDemo) directory to install the Leopard Cocoapod:
+
+```ruby
```
Comment (kenarsa) [inline demo/ios/README.md:18]: console?

**Review by kenarsa — CHANGES_REQUESTED** [review]

File: binding/ios/Leopard.swift (hunk 6 above)
Comment (ksyeo1010) [reply binding/ios/Leopard.swift:74]: yes, if you delete and run process.

**Review by kenarsa — APPROVED** [review]

**Pull request merged**